src/cs/ubc/ca/graphTeX/ast/BidirectionalEdge.java

```diff
@@ -3,8 +3,13 @@
 import cs.ubc.ca.graphTeX.visitor.Visitor;
 
 public class BidirectionalEdge extends Relation {
-    String firstNode;
-    String secondNode;
+    public String firstNode;
```

Comment:
Any reason these are public fields rather than private w/ getter/setter? Pretty indifferent either way, just curious.

Reply:
Yeah they can use getters/setters, it's just slightly more convenient to have them as public for when we use the visitor patterns to use their fields. We can change it if you feel strongly about it!